ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java

```diff
@@ -127,7 +133,11 @@ private SchemaResult fromParsedSchema(
 
     final List<SimpleColumn> columns;
     try {
-      columns = translator.toColumns(parsedSchema, isKey);
+      columns = translator.toColumns(
+          parsedSchema,
+          isKey,
+          serdeFeatures.enabled(SerdeFeature.UNWRAP_SINGLES)
```

Comment:
I'd probably just pass all the features down.  There could be other features later that also affect the schema.